ws-client/src/transport.rs

```diff
@@ -277,6 +267,31 @@ impl<'a> WsTransportClientBuilder<'a> {
 	}
 }
 
+impl<'a> TryFrom<WsConfig<'a>> for WsTransportClientBuilder<'a> {
+	type Error = WsHandshakeError;
+
+	fn try_from(config: WsConfig<'a>) -> Result<Self, Self::Error> {
+		let url =
+			url::Url::parse(&config.url).map_err(|e| WsHandshakeError::Url(format!("Invalid URL: {}", e).into()))?;
+		let mode = match url.scheme() {
+			"ws" => Mode::Plain,
+			"wss" => Mode::Tls,
+			_ => return Err(WsHandshakeError::Url("URL scheme not supported, expects 'ws' or 'wss'".into())),
+		};
+		let host = url.host_str().ok_or_else(|| WsHandshakeError::Url("No host in URL".into()))?.into();
+		let sockaddrs: Vec<SocketAddr> = url.socket_addrs(|| None).map_err(WsHandshakeError::ResolutionFailed)?;
```

Comment:
switched to https://docs.rs/url/2.2.0/url/struct.Url.html#method.socket_addrs instead which provides the default port number if it's missing such that we don't have to do it manually anymore

Reply:
Do you think we should add a comment stating this and what the default port is?

Reply:
I think the best option is the document this on `WsConfig` i.e, default ports is `ws - 80, wss - 433` ^^

Here we could add a comment that `sock_addrs()` uses Https://docs.rs/url/2.2.0/url/struct.Url.html#method.port_or_known_default if the port is missing